### PR TITLE
feat: Add charset  `v_func` for `StringBone` 

### PR DESCRIPTION
**Pull request by ArneGudermann**
Replacement for #1141

### CODE REVIEW

**Review by sveneberth — APPROVED** [review]

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, this looks good, can you please provide a docstring with an example as well?

File: src/viur/core/bones/string.py [hunk 1]
```diff
@@ -381,7 +381,16 @@ def structure(self) -> dict:
         return ret
 
     @classmethod
-    def v_func_valid_chars(cls, valid_chars=string.printable) -> t.Callable:
+    def v_func_valid_chars(cls, valid_chars: t.Iterable = string.printable) -> t.Callable:
+        """
+        Returns a function that takes a string and check whether it contains valid characters.
+        If all characters are valid, it returns None.
+        If not all characters are valid, it returns an error Message.
```
Comment (phorward) [inline src/viur/core/bones/string.py:388]:
```suggestion
        Returns a function that takes a string and checks whether it contains valid characters.
        If all characters of the string are valid, it returns None, and succeeds.
        If invalid characters are present, it returns an appropriate error message.
```

**Review by phorward — CHANGES_REQUESTED** [review]
@ArneGudermann can you provide an example definition for a `StringBone` that uses the `v_func_valid_chars`-feature? Then, developers can directly abstract the code for their specific use-case and the feature will be used with ease.

**Review by sveneberth — APPROVED** [review]

**Review by phorward — CHANGES_REQUESTED** [review]
Still missing an example in the docstring!

**Comment by ArneGudermann** [comment]
@phorward now ok ?

**Pull request merged**